src/TeaPie/Logging/RequestsLoggingHandler.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Net.Http.Headers;
+using TeaPie.Http;
+using TeaPie.Http.Auth;
+using Microsoft.Extensions.Logging;
+
+namespace TeaPie.Logging;
+
+internal class RequestsLoggingHandler(IAuthProviderAccessor authProviderAccessor, ILoggerFactory loggerFactory) : DelegatingHandler
+{
+    private readonly IAuthProviderAccessor _authProviderAccessor = authProviderAccessor;
+    private readonly ILogger _logger = loggerFactory.CreateLogger("HttpRequests");
+    private static readonly HttpRequestOptionsKey<RequestExecutionContext> _contextKey = new("__TeaPie_Context__");
+    public static readonly HttpRequestOptionsKey<RequestLogFileEntry> LogEntryKey = new("__TeaPie_LogEntry__");
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        if (!request.Options.TryGetValue(_contextKey, out var requestContext) || requestContext is null)
+        {
+            return await base.SendAsync(request, cancellationToken);
+        }
+
+        var logEntry = await GetOrCreateLogEntryAsync(request, requestContext);
+        var attemptStartTime = DateTime.UtcNow;
+        HttpResponseMessage? response = null;
+        Exception? exception = null;
+
+        try
+        {
+            response = await base.SendAsync(request, cancellationToken);
+            return response;
+        }
+        catch (Exception ex)
+        {
+            exception = ex;
+            throw;
+        }
+        finally
+        {
+            await RecordAttemptAsync(logEntry, response, exception, attemptStartTime);
+            await LogCompletedRequestAsync(request, response);
+        }
+    }
+
+    private async Task<RequestLogFileEntry> GetOrCreateLogEntryAsync(HttpRequestMessage request, RequestExecutionContext requestContext)
+    {
+        if (request.Options.TryGetValue(LogEntryKey, out var existingEntry) && existingEntry != null)
+        {
+            return existingEntry;
+        }
+
+        var logEntry = new RequestLogFileEntry
+        {
+            RequestId = Guid.NewGuid().ToString(),
```

Comment:
Ja by som tú `RequestId` spravil ako readonly. Lebo aj tak to Idčko generuješ už v konštruktore (pri definícií tej property v danej triede) a nikde inde nemá zmysel ho nastavovať